Java parser fails to handle empty enum

## What version of OpenRewrite are you using?

* org.openrewrite:rewrite-core:8.48.0

## What is the smallest, simplest way to reproduce the problem?

The Java parser does not seem to handle when a class contains an empty enum. In our code base this showed up as some imports in the class being removed by the RemoveUnusedImports recipe. Here's a minimised variant of the source code that triggered this for us:

```java
  @Test
  void importRemoval() {
    rewriteRun(
        spec -> spec.recipe(new RemoveUnusedImports()),
        // language=java
        java(
            """
            package com.helloworld;

            import java.time.Instant;

            public class InstallationStatus {

              public enum InstallationFeatures {}

              public static InstallationStatus create(Instant updateTime) {
                return null;
              }
            }
            """));
  }
```

When run, this fails with:

```
java.lang.AssertionError: Source file was parsed into an LST that contains non-whitespace characters in its whitespace. This is indicative of a bug in the parser.
package com.helloworld;

import java.time.Instant;

public class InstallationStatus {

  public enum InstallationFeatures {~~(non-whitespace)~~>}

  public static InstallationStatus create(Instant updateTime) {
    return null<~~;
  }
}
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:321)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:129)
```

When run in our production rewrites, this will remove the import of `java.time.Instant`, failing compilation.

--- a/rewrite-java-11/src/main/java/org/openrewrite/java/isolated/ReloadableJava11ParserVisitor.java
+++ b/rewrite-java-11/src/main/java/org/openrewrite/java/isolated/ReloadableJava11ParserVisitor.java
@@ -459,7 +459,9 @@ public class ReloadableJava11ParserVisitor extends TreePathScanner<J, Space> {
                     EMPTY
             );
         } else if (kind.getType() == J.ClassDeclaration.Kind.Type.Enum) {
-            if (positionOfNext(";", null) >= 0) {
+            int nextSemicolonPosition = positionOfNext(";", null);
+            int nextClosingBracePosition = positionOfNext("}", null);
+            if (nextSemicolonPosition >= 0 && nextSemicolonPosition < nextClosingBracePosition) {
                 enumSet = padRight(new J.EnumValueSet(randomId(), sourceBefore(";"), Markers.EMPTY, emptyList(), true), EMPTY);
             }
         }

--- a/rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17ParserVisitor.java
+++ b/rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17ParserVisitor.java
@@ -515,7 +515,9 @@ public class ReloadableJava17ParserVisitor extends TreePathScanner<J, Space> {
                     EMPTY
             );
         } else if (kind.getType() == J.ClassDeclaration.Kind.Type.Enum) {
-            if (positionOfNext(";", null) >= 0) {
+            int nextSemicolonPosition = positionOfNext(";", null);
+            int nextClosingBracePosition = positionOfNext("}", null);
+            if (nextSemicolonPosition >= 0 && nextSemicolonPosition < nextClosingBracePosition) {
                 enumSet = padRight(new J.EnumValueSet(randomId(), sourceBefore(";"), Markers.EMPTY, emptyList(), true), EMPTY);
             }
         }

--- a/rewrite-java-21/src/main/java/org/openrewrite/java/isolated/ReloadableJava21ParserVisitor.java
+++ b/rewrite-java-21/src/main/java/org/openrewrite/java/isolated/ReloadableJava21ParserVisitor.java
@@ -519,7 +519,9 @@ public class ReloadableJava21ParserVisitor extends TreePathScanner<J, Space> {
                     EMPTY
             );
         } else if (kind.getType() == J.ClassDeclaration.Kind.Type.Enum) {
-            if (positionOfNext(";", null) >= 0) {
+            int nextSemicolonPosition = positionOfNext(";", null);
+            int nextClosingBracePosition = positionOfNext("}", null);
+            if (nextSemicolonPosition >= 0 && nextSemicolonPosition < nextClosingBracePosition) {
                 enumSet = padRight(new J.EnumValueSet(randomId(), sourceBefore(";"), Markers.EMPTY, emptyList(), true), EMPTY);
             }
         }

--- a/rewrite-java-8/src/main/java/org/openrewrite/java/ReloadableJava8ParserVisitor.java
+++ b/rewrite-java-8/src/main/java/org/openrewrite/java/ReloadableJava8ParserVisitor.java
@@ -457,7 +457,9 @@ public class ReloadableJava8ParserVisitor extends TreePathScanner<J, Space> {
                     EMPTY
             );
         } else if (kind.getType() == J.ClassDeclaration.Kind.Type.Enum) {
-            if (positionOfNext(";", null) >= 0) {
+            int nextSemicolonPosition = positionOfNext(";", null);
+            int nextClosingBracePosition = positionOfNext("}", null);
+            if (nextSemicolonPosition >= 0 && nextSemicolonPosition < nextClosingBracePosition) {
                 enumSet = padRight(new J.EnumValueSet(randomId(), sourceBefore(";"), Markers.EMPTY, emptyList(), true), EMPTY);
             }
         }

--- a/rewrite-java-test/src/test/java/org/openrewrite/java/JavaParserTest.java
+++ b/rewrite-java-test/src/test/java/org/openrewrite/java/JavaParserTest.java
@@ -373,4 +373,22 @@ class JavaParserTest implements RewriteTest {
         assertThat(JavaParser.filterArtifacts("rewrite-java", classpath))
           .containsOnly(Paths.get("/.m2/repository/org/openrewrite/rewrite-java/8.41.1/rewrite-java-8.41.1.jar"));
     }
+
+    @Test
+    void emptyEnum() {
+        rewriteRun(
+          // language=java
+          java(
+            """
+            package com.helloworld;
+            import java.time.Instant;
+            public class InstallationStatus {
+              public enum InstallationFeatures {}
+              public InstallationStatus create(Instant updateTime) {
+                return null;
+              }
+            }
+            """));
+    }
+
 }

